gravitee-apim-console-webui/src/management/settings/top-apis/migrated/top-apis-dialog/top-apis-dialog.component.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, OnInit } from '@angular/core';
+import { MatDialogRef } from '@angular/material/dialog';
+import { FormControl } from '@angular/forms';
+import { forkJoin, Observable, Subject } from 'rxjs';
+import { filter, map, startWith, takeUntil, tap } from 'rxjs/operators';
+import { filter as lodashFilter, includes as lodashIncludes, map as lodashMap } from 'lodash';
+
+import { TopApi } from '../top-apis.model';
+import { ApiService } from '../../../../../services-ngx/api.service';
+import { Api } from '../../../../../entities/api';
+import { TopApiService } from '../../../../../services-ngx/top-api.service';
+
+@Component({
+  selector: 'app-top-apis-dialog',
+  templateUrl: './top-apis-dialog.component.html',
+  styleUrls: ['./top-apis-dialog.component.scss'],
+})
+export class TopApisDialogComponent implements OnInit {
+  private unsubscribe$: Subject<void> = new Subject<void>();
+  public searchApiControl: FormControl<string | Api> = new FormControl('');
+  public apis: Api[] = [];
+  public filteredOptions$: Observable<Api[]>;
+  public isApiSelected = false;
+
+  constructor(
+    public dialogRef: MatDialogRef<TopApisDialogComponent>,
+    private apiService: ApiService,
+    private topApiService: TopApiService,
+  ) {}
+
+  ngOnDestroy() {
+    this.unsubscribe$.next();
+    this.unsubscribe$.complete();
+  }
+
+  ngOnInit(): void {
+    forkJoin([this.apiService.getAll(), this.topApiService.getList()])
```

Comment:
Do we really need to get the big long list of APIs? What if we used the functionality used with the main API List and its search bar?